packages/secure_storage/amplify_secure_storage_dart/lib/src/utils/file_key_value_store.dart

```diff
@@ -47,9 +53,11 @@ class FileKeyValueStore {
     required String key,
     required Object value,
   }) async {
-    final data = await readAll();
-    data[key] = value;
-    return writeAll(data);
+    return _scheduler.schedule(() async {
+      final data = await readAll();
+      data[key] = value;
+      return writeAll(data);
+    });
   }
 
   /// Overwrites the existing data.
```

Comment:
Should writeAll be wrapped by the scheduler as well?

Reply:
Discussed offline - I updated writeAll and readAll to be annotated with `@visibleForTesting` as these apis are really private and only exposed for testing.